### ISSUE TITLE
Job delayed too long, service full error when bulk uploading

### ISSUE THREAD

**Issue by BonaBeavis** [dathere/datapusher-plus#79]
As you can see, I am putting your plugin through the paces ;) When I submit all the datasets at once, some get this error message:

![image](https://user-images.githubusercontent.com/3330855/231837407-44b43283-d180-4c23-92db-0d8d8c61da58.png)

When I retry, it works. So I would guess there is some timeout set too low.

**Comment by jqnatividad** [comment]
Thanks @BonaBeavis for pushing it ;)

Will investigate further.  Can you share more info about the number of jobs enqueued? Were they local files? URLs? How big were the files?

**Comment by BonaBeavis** [comment]
About 60 datasets, both local and remote. The datasets are not big, about a few dozen entries and a dozen headers.

I am running `ckan datapusher submit --yes` do you have some command I could run to get a debug output?

**Comment by jqnatividad** [comment]
Yes @BonaBeavis , you can turn on logging by tweaking some values in your `.env` file (https://github.com/dathere/datapusher-plus/blob/master/datapusher/dot-env.template)

You can also have qsv create its own logs by setting these env_vars:

```
export QSV_LOG_LEVEL=debug
export QSV_LOG_DIR=/tmp
```

The log file should be named `qsvdp_rCURRENT.log` in /tmp.

**Comment by jqnatividad** [comment]
This will be fixed with DP+ 1.x, where we convert DP+ to be an extension, rather than a separate web service.

cc @tino097

**Comment by BonaBeavis** [comment]
Workaround until then:

```bash
for id in $(ckan dataset list | grep -oP '^\w{8}-\w{4}-\w{4}-\w{4}-\w{12}')
do
  ckan datapusher submit --yes $id
  sleep 1
done
```

**Comment by Gutts-n** [comment]
Hello guys, any update on this issue? I'm facing the same problem.

**Comment by tino097** [comment]
> Hello guys, any update on this issue? I'm facing the same problem.

Are you using the DP+ as extension  or still as service?

**Comment by Gutts-n** [comment]
@tino097 when I was facing this problem, it was a service. 

But now using it as an extension, I click to upload the file, and it never starts the job. This is presented on the logs for each click on the "upload to datastore" button:

![Image](https://github.com/user-attachments/assets/500aae1e-540c-44ae-8980-b56a9bc2708f)

**Comment by tino097** [comment]
You will need to run the worker using the command  
`ckan jobs worker`

**Comment by Gutts-n** [comment]
@tino097 it "worked", it has finished the job, but on the task_status table it's pending yet, also I don't have any logs on the datapusher page, and the file is not present on the datastore, so in short it's not doing anything inside of the job, only showing success, I'm using the 2.0.0 version.

The problem is that there is no log informing what is happening inside of the job :(

![Image](https://github.com/user-attachments/assets/c94d58c6-2ef3-454c-b9db-88754db1d07e)

![Image](https://github.com/user-attachments/assets/42d31444-2dce-4b47-9c42-64474a24ac31)

cc: @jqnatividad

**Comment by Gutts-n** [comment]
@tino097 hey, I ran the command to upgrade the db, but I'm still receiving nothing in the logs. A thing that I forgot to share is that there is this log before it finishes the job:

![Image](https://github.com/user-attachments/assets/8ca35734-84f0-41b9-ac8e-75284e5c2e38)

Also, do I need to have the scheming installed in the environment for the extension to work?

**Comment by Gutts-n** [comment]
@tino097 I've found the problem, the qsv wasn't on the path of the system that was running the CKAN, so it wasn't able to run the qsv command, I think that would be nice to have a log informing it when it happens, the job just crashed and finished without any information about that, I could fix it basically in trial and error.

**Comment by jqnatividad** [comment]
Hi @Gutts-n , DP+ does check if the qsv binary is available and logs it in the datastore tab.

Will definitely add an installation test after a fresh install or upgrade to ensure DP+ is installed properly.

**Comment by Gutts-n** [comment]
@jqnatividad thanks for the reply. 

It's almost running here, I'd like to check if this step takes this time to run:

![Image](https://github.com/user-attachments/assets/cd20574d-1a4c-41a1-934a-187babcb6aff)

It's a file of 1GB, and there are no new logs related to the job.

**Comment by Gutts-n** [comment]
Hey @jqnatividad , I tested the extension in a container, and I noticed that the container crashed with an out of memory error when this step started to run, then I increased the memory allocated to it, and it fixed. Thank you for the help.
